### ISSUE TITLE
Huge Discrepancy in GCAM v6 and GCAM v7- oil primary energy consumption for US

### ISSUE THREAD

**Issue by mchauhan2383** [JGCRI/gcam-core#327]
Hi all, While working with the reference scenario in both GCAM v6 and GCAM v7- I have found huge difference in the Oil Primary Energy consumption for the historical and the forecast years for USA. The values in v6 looks more reasonable and trend closely with other publicly available data- while the values in v7 are very low. Are there any reasons as to why the values are differing so much for the historicals ?

GCAM v7

scenario                                                        | region | fuel  | 1990      | 2005     | 2010      | 2015      | 2020       | 2025   | 2030    | 2035
Reference,date=2023-31-7T09:21:42-05:00 | USA     | a oil | 18.2167 | **13.5568 | 15.0209 | 24.0397 | 22.5693** | 20.717 | 18.041 | 16.4806

GCAM v6

scenario                                                      | region | fuel | 1990       | 2005      |  2010     | 2015   | 2020       | 2025    | 2030     | 2035
Reference,date=2023-6-4T05:27:50-05:00 | USA     | a oil | 34.0364 | **40.8455 | 35.8874 | 35.717 | 36.2195** | 35.0434 | 32.465 | 30.2376

**Comment by pralitp** [comment]
@mchauhan2383 which query did you use?  

I just checked using the query "primary energy consumption with CCS by region (direct equivalent)" and verified GCAM 6 and 7 match in the historical years (at the values you quote for GCAM 6).

However using the query "primary energy consumption by region (direct equivalent)" replicates the issue you are observing where GCAM 7 values are too low.  This query relies on some meta data to be passed through to the output for querying and I'm thinking we must have missed some of the meta data in some change in GCAM 7 resulting in invalid query results.

We'll get that cleaned up.  In the mean time maybe you can switch to using the "primary energy consumption with CCS by region (direct equivalent)" (which unfortunately takes significantly longer to run)

**Comment by pkyle** [comment]
Not sure if this help @pralitp but the numbers being reported for oil by the primary energy queries right now are equal to the "crude oil" input to "regional crude oil," represents consumption of own (domestic) oil production. It excludes consumption of imported oil.

**Comment by mchauhan2383** [comment]
@pralitp Yes, I used the query primary energy consumption by region (direct equivalent)

**Comment by mchauhan2383** [comment]
@pralitp this query is taking a lot of time in my machine (its still running after half an hour). Are there ways to speed up the performance of this query ?

**Comment by mchauhan2383** [comment]
Facing this issue as well-

Stopped at ., 22/96:[XPTY0004] Cannot promote empty-sequence() to node(): $currTree := (). Stack Trace:- ., 167/61- ., 174/30

**Comment by mchauhan2383** [comment]
@pkyle @pralitp Unable to extract data using the query **Primary energy consumption with CCS by region (direct equivalent)**
![image](https://github.com/JGCRI/gcam-core/assets/59660019/abd072f6-8c88-4530-81e0-f70e21348a10)
Could you please help me in resolving this ?

**Comment by pkyle** [comment]
In my own work I generally don't use those primary energy queries; I don't understand the CDATA code all that well, and given the fuel-specific accounting conventions that go into primary energy reporting, I prefer to manually perform whatever calculations need to be done in order to translate from the model's inter-sectoral flows to the reported primary energy consumption. It's not that complex; there are generally four categories that I add up: (1) standard upstream fossil+bio (sectors: "regional coal", "regional oil", "regional natural gas", "regional biomass"), (2) non-fossil electricity generation (electricity subsectors: "nuclear", "wind", "solar", "hydro", "geothermal"), (3) non-fossil hydrogen (H2 central production subsectors: "nuclear", "wind", "solar"), and (4) first-generation biofuels (sectors: "regional corn for ethanol", "regional biomassOil"). A simple query like "outputs by subsector" should return all of the information needed to post-process the data into the relevant categories for calculating primary energy consumption using the "direct equivalent" accounting convention.
